src/strands/experimental/bidirectional_streaming/agent/agent.py

```diff
@@ -47,117 +46,12 @@ class BidirectionalAgent:
-                tool_use: ToolUse = {
-                    "toolUseId": tool_id,
-                    "name": normalized_name,
-                    "input": kwargs.copy(),
-                }
-                tool_results: list[ToolResult] = []
-                invocation_state = kwargs
-
-                async def acall() -> ToolResult:
-                    async for event in ToolExecutor._stream(self._agent, tool_use, tool_results, invocation_state):
-                        _ = event
-
-                    return tool_results[0]
-
-                def tcall() -> ToolResult:
-                    return asyncio.run(acall())
-
-                with ThreadPoolExecutor() as executor:
-                    future = executor.submit(tcall)
-                    tool_result = future.result()
-
-                # Always record direct tool calls for bidirectional agents to maintain conversation history
-                # Use agent's record_direct_tool_call setting if not overridden
-                if record_direct_tool_call is not None:
-                    should_record_direct_tool_call = record_direct_tool_call
-                else:
-                    should_record_direct_tool_call = self._agent.record_direct_tool_call
-
-                if should_record_direct_tool_call:
-                    # Create a record of this tool execution in the message history
-                    self._agent._record_tool_execution(tool_use, tool_result, user_message_override)
-
-                return tool_result
-
-            return caller
-
-        def _find_normalized_tool_name(self, name: str) -> str:
-            """Lookup the tool represented by name, replacing characters with underscores as necessary."""
-            tool_registry = self._agent.tool_registry.registry
-
-            if tool_registry.get(name, None):
-                return name
-
-            # If the desired name contains underscores, it might be a placeholder for characters that can't be
-            # represented as python identifiers but are valid as tool names, such as dashes. In that case, find
-            # all tools that can be represented with the normalized name
-            if "_" in name:
-                filtered_tools = [
-                    tool_name for (tool_name, tool) in tool_registry.items() if tool_name.replace("-", "_") == name
-                ]
-
-                # The registry itself defends against similar names, so we can just take the first match
-                if filtered_tools:
-                    return filtered_tools[0]
-
-            raise AttributeError(f"Tool '{name}' not found")
-
     def __init__(
         self,
-        model: BidirectionalModel,
-        tools: list | None = None,
-        system_prompt: str | None = None,
-        messages: Messages | None = None,
+        model: Union[BidirectionalModel, str, None] = None,
+        tools: Optional[list[Union[str, dict[str, str], Any]]] = None,
```

Comment:
Lets not copy the Agent class here, dont love how it passes in tools. I think we should do something like this:
```suggestion
        tools: list[string, AgentTool, ToolProvider] = None,
```